src/pages/changes.tsx

```diff
@@ -0,0 +1,106 @@
+    nodes: ChangelogEntry[],
+  }
+};
+
+type ChangelogEntry = {
+  title: string,
+  date: string, // e.g. "2019-10-01"
+  body: {
+    json: Document,
+  },
+  node_locale: string,
+};
+
+type YearAndMonth = {year: number, month: number, localeString: string};
+
+/**
+ * Given a date string in the form "YYYY-MM-DD", returns the "YYYY-MM" part.
+ */
+const isSameYearAndMonth = (a: YearAndMonth, b: YearAndMonth) => a.year === b.year && a.month === b.month;
+
+const parseYearAndMonth = (date: string, locale: string): YearAndMonth => {
+  const year = parseInt(date.slice(0, 4), 10);
+  const month = parseInt(date.slice(5, 7), 10);
+  const dateObj = new Date(year, month - 1);
+  const localeString = dateObj.toLocaleString(locale, {month: 'long', year: 'numeric'});
+
+  return {year, month, localeString};
+};
+
+const ChangelogEntry: React.FC<{node: ChangelogEntry, prevNode?: ChangelogEntry}> = ({node, prevNode}) => {
+  const currYM = parseYearAndMonth(node.date, node.node_locale);
+  const prevYM = prevNode ? parseYearAndMonth(prevNode.date, prevNode.node_locale) : undefined;
+  const showYM = !prevYM || !isSameYearAndMonth(currYM, prevYM);
+
+  return <>
+    {showYM && <h2 className="date-and-month">{currYM.localeString}</h2>}
+    <h3>{node.title}</h3>
+    {documentToReactComponents(node.body.json)}
+  </>;
+};
+
+export const Changelog: React.FC<{entries: ChangelogEntries}> = ({entries}) => {
+  const { nodes } = entries.allContentfulChangelogEntry;
+
+  return (
+    <Layout>
+      <div className="changes-page">
+        <section className="hero is-small">
+          <div className="hero-body has-text-centered is-horizontal-center">
+            <div className="container">
+              <h1 className="title is-size-2 has-text-grey-dark has-text-weight-normal is-spaced">
+                Changelog
+              </h1>
+              <div className="subtitle has-text-grey-dark is-italic">
+                Learn what's new in JustFix.nyc's products
+              </div>
+            </div>
+          </div>
+          <div className="hero-body is-horizontal-center">
+            <div className="content has-text-grey-dark">
+            {nodes.map((node, i) => (
+              <ChangelogEntry node={node} prevNode={i > 0 ? nodes[i - 1] : undefined} />
+            ))}
+            </div>
+          </div>
+        </section>
+      </div>
+    </Layout>
+  );
+};
+
+export const LocalizedChangelogEntries = graphql`
+fragment LocalizedChangelogEntries on Query {
+  allContentfulChangelogEntry(
+    filter: {node_locale: {eq: $locale}},
+    sort: {order: DESC, fields: [date, title]}
+  ) {
+    nodes {
+      title
+      date
+      body {
+        json
+      }
+      node_locale
+    }
+  }
+}
+`;
+
+const EnglishChangelogPage: React.FC<{}> = () => (
+  <StaticQuery query={graphql`
+  query ($locale: String! = "en-US") { ...LocalizedChangelogEntries }
+  `} render={data => (<Changelog entries={data} />)} />
+);
```

Comment:
And lastly, other pages have used a render function something like `{data => (<TeamPageScaffolding content={data.contentfulTeamPage} />}`. I'm fine with keeping what you wrote, but just wanted to flag the standard I've used for other pages.

Reply:
So I was thinking of doing this, but I actually wanted to keep as much of the response as strongly-typed as possible (and keep things in the other locales as DRY as possible) which is why I ended up typing the whole `data` blob and passing it wholesale to the `Changelog` component.  I can definitely rename the `entries` prop to be `content` so that part adheres to convention though!

Reply:
Oh another advantage of passing the query response wholesale is that it's possible we might actually want to make multiple queries in that single GraphQL query, e.g. `allContentfulChangelogEntry` along with `allContentfulProductCategories` (the second one is just made up)... if we just have one big strongly-typed data blob we pass in then we won't have to split it up into multiple props, which makes things less DRY on our localized pages too... anyhow just some thoughts.

Reply:
Ok I renamed the main prop to `content` in c29d8d5a!

Reply:
oh right, that's a good point! Yeah for the other pages, there's one-to-one parity with a single Contentful model so I developed that convention of saving just the specific data blob we need as a prop within the scaffolding component. But definitely makes sense that there will be cases where we may be querying multiple different Contentful models in one graphql query.